src/record/import/repositories/parsers/parseCsv/index.ts

```diff
@@ -0,0 +1,61 @@
+import type { RecordSchema } from "../../../types/schema";
+
+import csvParse from "csv-parse";
+
+import { convertRecord, recordReader } from "./record";
+import { SEPARATOR } from "./constants";
+import { ParserError } from "../error";
+import type { LocalRecordRepository } from "../../../usecases/interface";
+
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions#use_of_the_yield_keyword
+// eslint-disable-next-line func-style
+export async function* csvReader<T extends NodeJS.ReadableStream>(
+  source: () => T,
```

Comment:
Type [`NodeJS.ReadableStream`](https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules__types_node_globals_d_.nodejs.readablestream.html) is an interface.  You don't have to use a generics:

```suggestion
export async function* csvReader(
  source: () => NodeJS.ReadableStream,
```

Reply:
fixed